libtenzir/include/tenzir/arrow_table_slice.hpp

```diff
@@ -128,18 +128,17 @@ class arrow_table_slice final {
 // -- utility functions -------------------------------------------------------
 
 /// Access a Tenzir data view for a given row in an Arrow Array.
-data_view value_at(const type& type, const std::same_as<arrow::Array> auto& arr,
-                   int64_t row) noexcept;
+auto value_at(const type& type, const std::same_as<arrow::Array> auto& arr,
```

Comment:
Was the `noexcept` drop intentional here and below?

Reply:
Yes. We have some asserts in there, and they are throwing. So annotating the function `noexcept` essentially just adds a `catch (...) { std::terminate() }` there, which we definitely do not want to have. In general, we are moving away from using `noexcept` everywhere, and I converted these functions as part of adding an overload.

Reply:
Yes—we recently switched to having our assertions throw. Given that the engine is actor-based we install an exception handler at every actor to be able to shut down a reasonable subset of the system when an exception occurs, instead of shutting down everything. So now we have some places in which we historically had noexcept that we're now migrating away from noexcept opportunistically.

Reply:
For some reason your comment didn't show up earlier for me, and I was late to the party apparently. 😅